ary/models.py

```diff
@@ -279,3 +283,185 @@ def can_modify(self, user):
         )
+
+    def _load_schema(self, FieldClass, GroupClass=None):
+        schema = {}
+        assessment_template = self.lead.project.assessment_template
+        groups = GroupClass.objects.filter(template=assessment_template)
+        schema = {
+            group.title: [
+                {
+                    'id': field.id,
+                    'name': field.title,
+                    'type': field.field_type,
+                    'options': {
+                        x['key']: x['title'] for x in field.get_options()
+                    }
+                }
+                for field in group.fields.all()
+            ] for group in groups
+        }
+        return schema
+
+    def _data_from_schema(self, schema, raw_data):
+        if not raw_data:
+            return {}
+        if 'id' in schema:
+            key = str(schema['id'])
+            value = raw_data.get(key, '')
+            if schema['type'] == Field.SELECT:
+                value = schema['options'].get(value, value)
+            if schema['type'] == Field.MULTISELECT:
+                value = [schema['options'].get(x, x) for x in value]
+            return {schema['name']: value}
+        if isinstance(schema, dict):
+            data = {
+                k: self._data_from_schema(v, raw_data)
+                for k, v in schema.items()}
+        elif isinstance(schema, list):
+            data = [self._data_from_schema(x, raw_data) for x in schema]
+        else:
+            raise Exception("Something that could not be parsed from schema")
+        return data
+
+    def get_metadata_json(self):
+        metadata_schema = self._load_schema(MetadataField, MetadataGroup)
+        metadata_raw = self.metadata or {}
+        metadata_raw = metadata_raw.get('basic_information', {})
+        metadata = self._data_from_schema(metadata_schema, metadata_raw)
+        return metadata
+
+    def get_methodology_json(self):
+        methodology_sch = self._load_schema(MethodologyField, MethodologyGroup)
+        methodology_raw = self.methodology or {}
+
+        mapping = {
+            'attributes': lambda x: self._data_from_schema(methodology_sch, x),
+            'sectors': lambda x: Sector.objects.get(id=x).title,
+            'focuses': lambda x: Focus.objects.get(id=x).title,
+            'affected_groups': lambda x: x['name'],
+            'locations': lambda x: GeoArea.objects.get(id=x).title,
+            'objectives': identity,
+            'sampling': identity,
+            'limitations': identity,
+            'data_collection_techniques': identity
+        }
+
+        return {
+            underscore_to_title(k):
+                v if not isinstance(v, list)
+                else [mapping[k](y) for y in v]
+            for k, v in methodology_raw.items()
+        }
+
+    def get_summary_json(self):
+        # functions to get exact value of an entry in summary group
+        def identity(x):
```

Comment:
Duplicate `identity` function that is already defined in *utils/common.py*.

Reply:
Roger That